### PR TITLE
feat: add feed form e2e

### LINKED ISSUE #658: Submit Form Cypress test full flow
Create Cypress tests that cover full flow of submitting a form

### PR DESCRIPTION
**Pull request by Alessandro100**
closes #658 

**Summary:**

Added e2e tests for the add feed form flow and fixed login bug

**Expected behavior:** 

All the e2e tests should pass

**Testing tips:**

Run the e2e tests

Terminal 1: `yarn run start:dev`
Terminal 2: `yarn run cypress:open`

GUI will open where you will be able to run the e2e tests

Please make sure these boxes are checked before submitting your pull request - thanks!

** Learnings **
E2e tests are slow, we should be conscious of what kind of tests we write. It should cover the main paths of a feature and not all the edge cases.

- [ ] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [ ] Add or update any needed documentation to the repo
- [ ] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

image

### CODE REVIEW

File: web-app/src/app/store/saga/auth-saga.ts [hunk 1]
```diff
@@ -191,11 +191,13 @@ function* resetPasswordSaga({
...
 function* anonymousLoginSaga(): Generator {
   try {
+    const auth = getAuth();
+    const firebaseUser = app.auth().currentUser;
     // Check if the user is already authenticated
     const isAuthenticated: boolean = (yield select(
       selectIsAuthenticated,
     )) as boolean;
-    if (isAuthenticated) {
+    if (isAuthenticated && firebaseUser !== null) {
```
Comment (Alessandro100) [inline web-app/src/app/store/saga/auth-saga.ts:200]: @davidgamez This was added because there are certain cases during the tests where there was an authenticated user but no firebase user. This lead to the App being stuck in a state of not ready which leads to having just a white screen

I suspect this is the fix, and that it will fix the ResetPassword cypress test as well

Comment (davidgamez) [reply web-app/src/app/store/saga/auth-saga.ts:200]: Great finding!

File: web-app/cypress/support/commands.ts [hunk 2]
```diff
@@ -36,3 +36,35 @@
...
+      payload: {
+        fullName: 'Valery',
+        email: 'testuser@gmail.com',
+        isRegistered: true,
+        isEmailVerified: true,
+        organization: '',
+        isRegisteredToReceiveAPIAnnouncements: false,
+        isAnonymous: false,
+        refreshToken:
+          'AMf-vBwvDFwWA77IKuTUdQ9eZ7sCalLb3LPfHupyvvI91SYcImb_e5R417gPIZbVJxaJUSvMDqHWlQaMMZQZkkahT9zFW3FXymDMmvSzKB-NKO2X_lw1yCP_YyegslW6Wl4y2PRG_gEyUHxpESrjDVI_scIDIvfHaoH97-GraKJFgmCo61QzmYyMY8vyaOxyH9ovcVwdiy-KLpRbU-B8VrtrRwKg-o7BWggTlIRXu5hgtG34lS8GCdM',
```
Comment (davidgamez) [inline web-app/cypress/support/commands.ts:54]: Is this a _valid_ dev refresh token? If it is, we must have it as a variable injected by the ci instead of plain text.
